deps/src/matrices.cpp

```diff
@@ -22,14 +28,28 @@ void singular_define_matrices(jlcxx::Module & Singular)
 
     Singular.method("mp_Sub", &mp_Sub);
 
+    Singular.method("mp_Transp", &mp_Transp);
+
     Singular.method("mp_Mult", &mp_Mult);
 
+    Singular.method("mp_MultP", &mp_MultP);
+
     Singular.method("mp_Equal", &mp_Equal);
 
+    Singular.method("mpNew", [](ring R, int r, int c) {
+        rChangeCurrRing(R);
```

Comment:
The current ring should be changed back to the original after setting it. See other functions that do this to see how it's done.

Reply:
The ChangeRing seems useless, I remove it in the next update.